### DISCUSSION TITLE
Training Frigate+ for Vehicle detection in Black & White / IR at night

### DISCUSSION THREAD

**Discussion by gadget-man** [blakeblackshear/frigate#9098] [General]
Hi, I've seen several posts here about the problems with vehicle / object detection at night due to the existing models not being trained for black & white. I've now subscribed to Frigate+ (awesome for license plate detection during the day, in conjunction with PlateRecognizer for automated gate opening), but I now want to start training my model for vehicle detection at night.  The problem I'm having is that because frigate detects motion, but isn't able to detect any objects, I don't have any saved snapshots to upload to Frigate. 

What is the best way to capture snapshots so that I can start uploading them and training them into my Frigate+ model?

Thanks!

**Comment by NickM-27** [comment]
have you generated a frigate+ model yet? it should work at least reasonably well at night, we have a user that runs ALPR with frigate+ and it detects at night after they made motion detection sensitive enough

**Comment by moandcompany** [comment]
Semi-related:  Have you tried using the 'improve contrast' option in your Frigate Configuration to see if it helps with your night-time detections? (reference:  https://docs.frigate.video/configuration/)

**Comment by gadget-man** [comment]
Yes I've got my first Frigate+ model running (although only trained on about 50 images, almost all daytime ones), and I do have improve_contrast turned on.  I increased motion sensitivity slightly and that does seem to have helped a bit. I'll continue to monitor.

**Comment by kyle4269** [comment]
Here's an example of my night detection. As Nick said, I did have to crank my motion threshold up to 85 at night.  But frigate does detect the car and plate. 

 ![image](https://github.com/blakeblackshear/frigate/assets/18011470/e0bb320e-c4bf-4c45-963f-4a6ff4f6a1f9)

**Reply by gadget-man** [reply]
By setting motion threshold at 85, that means you've made motion detection significantly less sensitive at night time? (or that's how I read the docs on configuration). That's the opposite of what I was expecting - I thought I'd need to set motion detection more sensitive (i.e. a lower number) at night.

**Reply by kyle4269** [reply]
I just ran a clip through for testing.. I set my threshold to 1(which should be the most sensitive by the documents) and got zero motion boxes which resulted in no detection. I then set it back to 85 and received motion boxes plus car/plate detection.
```
    motion:
      threshold: 85
      improve_contrast: true
      contour_area: 30
```

**Reply by NickM-27** [reply]
this is going to be a product specifically of your camera. Due to there being so little contrast except on small areas the improve contrast is doing a lot of work. This then triggers the lightning threshold because basically the entire image has motion. Setting a lower threshold fixes this.

@gadget-man in your case this would not apply unless your image is similarly low contrast

**Reply by kyle4269** [reply]
Thanks for the explanation Nick! I have been wondering why it seemed 85 was more sensitive.

**Reply by bracklanna** [reply]
Seeking clarification re: Frigate+ and night/B&W vehicle ("car") object detection in general, not LPR specifically:

If I sign up for Frigate+ will my system immediately get better at recognizing vehicles at night, due to the overall improved training set for night/B&W car footage in general?

Or will I have to submit my own training data specifically from my cameras in order to get improved night/B&W vehicle object detection?

(My setup:  Frigate 0.16.2-4d58206, 6th gen intel core i5-6600, 16GB RAM, intel-vaapi GPU, Coral USB TPU,  6 x ANPVIZ® 5MP 1/3" cmos sensor POE cameras)

**Reply by hawkeye217** [reply]
Frigate+ has many black and white/IR images in the base model compared to the stock free model based on the COCO dataset, so your experience should improve immediately.

**Reply by bracklanna** [reply]
Confirmed after subscribing to Frigate+. 

The stock model that come with Frigate is basically useless for night detection. Subscribing to Frigate+ and switching (in my case) to mobildet 320x320 immediately brought night detection accuracy up by a huge margin, at least an order of magnitude, maybe closer to 2 orders. 

Question -- why does the mobiledet model show a train date of 11/14/2025 when that date hasn't come yet? Is that an error or some sort of programmatic date offset I'm not understanding?

![Screenshot 2025-10-31 at 9 06 13 PM](https://github.com/user-attachments/assets/1be4b6c7-7607-486b-8258-e014058294f4)

**Reply by blakeblackshear** [reply]
That's just an error on the backend side. Thanks for pointing it out.
